src/dune_engine/string_with_vars.mli

```diff
@@ -57,7 +55,9 @@ val has_pforms : t -> bool
 val text_only : t -> string option
 
 module Mode : sig
-  (** Expansion may produce either a [Single] value or [Many] values
+  (** Expansion may produce either a [Single] value or [Many] values.
+      [At_least_one] is for cases where at least one value is required, such as
+      in [(run %{foo})].
```

Comment:
This paragraph describing Single, Many, and At_least_one is better split and attached to the individual constructors below. E.g.

```
type _ t =
  | Single : Value.t t (** ... *)
  ..
```

Reply:
Sounds good, done